aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1984,4 +1985,29 @@ Tensor movedim(const Tensor& self, int64_t src, int64_t dst) {
   return at::movedim(self, IntArrayRef{src}, IntArrayRef{dst});
 }
 
+Tensor trace_cpu(const Tensor& self) {
```

Comment:
@zou3519 I think we should update this to be a composite of `diag` and `sum` to remove the type promotion inconsistencies. Obviously, we should benchmark the performance and keep the existing implementation if it's favorable.

Reply:
I benchmarked the performance of diag() and sum() a while back. It turns out that diag() + sum() is slower than the original TH implementation which is why the code looks like how it does today :/